### PR TITLE
Added the view abstract class

### PR DESCRIPTION
**Pull request by WakkaTrout**
Closes #(issue)

### Key features
* Created abstract class for views

### Future Improvements
* None

### Notes
Need to make pull request so I can get that navbar

### CODE REVIEW

File: src/views/abstractview.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class View: public QWidget
+{
+    Q_OBJECT
+public:
+
+    explicit View(QWidget* parent = nullptr)
+        : QWidget(parent)
+    {}
+    virtual ~View(){}
+    virtual void resetView() = 0;
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Docs for this function

File: src/views/abstractview.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * @class AbstractViewModel class
+ *
+ * An abstract base class that defines 2 basic functions that
+ * views must define: resetView() and resetUI()
+ */
+class View: public QWidget
+{
+    Q_OBJECT
+public:
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Let's move this a line down
```cpp
    Q_OBJECT

public:
    explicit View...
```

File: src/views/abstractview.hpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    explicit View(QWidget* parent = nullptr)
+        : QWidget(parent)
+    {}
+    virtual ~View(){}
+    virtual void resetView() = 0;
+
+    /**
+     * Resets the UI that corresponds with the given view
+     */
+    virtual void resetUI() = 0;
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Camel casing: `resetUi`

**Review by ChrisRzech — CHANGES_REQUESTED** [review]
The abstract class works like I expected, nice. The only thing you should do is remove `TeamView`. Each view should be completed in its own PR; while this PR should just handle the abstract view. Also, Roger is working on the team listing widget, so your current `TeamView` will be revamped.

**Review by Roger-Chavez — APPROVED** [review]

File: GrayonSlam.pro [hunk 4]
```diff
@@ -35,11 +35,13 @@ SOURCES += \
     src/datastore/team.cpp \
     src/datastore/souvenir.cpp
 
+
```
Comment (ChrisRzech) [inline GrayonSlam.pro]: woah woah, you thought you could pull a fast one on me?

File: GrayonSlam.pro [hunk 5]
```diff
@@ -48,7 +50,7 @@ HEADERS += \
 
 FORMS += \
     src/windows/mainwindow.ui \
-    src/windows/login.ui \
+    src/windows/login.ui
```
Comment (ChrisRzech) [inline GrayonSlam.pro]: Uncompiling change, you didn't try and run it?

File: src/views/abstractview.hpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <QWidget>
+
+/**
+ * @class AbstractViewModel class
+ *
+ * An abstract base class that defines 2 basic functions that
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Forgot period at the end of sentence. (yeah ik very picky, but documentation should be complete sentences)

File: src/views/abstractview.hpp [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+ * views must define: resetView() and resetUI()
+ */
+class View: public QWidget
+{
+    Q_OBJECT
+    
+public:
+    explicit View(QWidget* parent = nullptr)
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: A view should always have a parent, so putting a default parameter shouldn't happen.

File: src/views/abstractview.hpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        : QWidget(parent)
+    {}
+    virtual ~View(){}
+    /**
+     * Resets any internal components of the view
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Incomplete sentence

File: src/views/abstractview.hpp [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+     */
+    virtual void resetView() = 0;
+
+    /**
+     * Resets the UI that corresponds with the given view
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: Incomplete sentence

File: src/views/abstractview.hpp [hunk 10]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <QWidget>
+
+/**
+ * @class AbstractViewModel class
+ *
+ * An abstract base class that defines 2 basic functions that
+ * views must define: resetView() and resetUI()
```
Comment (ChrisRzech) [inline src/views/abstractview.hpp]: `resetUI()` doesn't match a function in `View`; however, `resetUi()` does.

**Review by ChrisRzech — CHANGES_REQUESTED** [review]
It doesn't look like this compiles. You are missing a `\` for line 50 in the `.pro` file.

File: src/views/abstractview.hpp (hunk 7 above)
Comment (WakkaTrout) [reply src/views/abstractview.hpp]: I just used the format that qt has so blame qt

Comment (ChrisRzech) [reply src/views/abstractview.hpp]: Fricken Qt 👎

Comment (WakkaTrout) [reply src/views/abstractview.hpp]: So no default then?

Comment (ChrisRzech) [reply src/views/abstractview.hpp]: No default
```cpp
QWidget* parent
```

**Comment by WakkaTrout** [comment]
Alright everything changed to how you want it, Chris

**Review by ChrisRzech — APPROVED** [review]
👍

**Pull request merged**